megamek/src/megamek/client/bot/BotClient.java

```diff
@@ -112,24 +119,40 @@ public void gamePlayerChat(GamePlayerChatEvent e) {
 
             @Override
             public void gameTurnChange(GameTurnChangeEvent e) {
-                if (isMyTurn()
-                        && (e.getPlayer() != null) 
-                        && (e.getPlayer().getId() == localPlayerNumber)) {
+                // On simultaneous phases, each player ending their turn will generalte a turn change
+                // We want to ignore turns from other players and only listen to events we generated
+                boolean ignoreSimTurn = game.isPhaseSimultaneous() && (e.getPreviousPlayerId() != localPlayerNumber)
+                        && calculatedTurnThisPhase;
+
+                
+                if (isMyTurn() && !ignoreSimTurn) {
+                    calculatedTurnThisPhase = true;
                     // Run bot's turn processing in a separate thread.
                     // So calling thread is free to process the other actions.
                     Thread worker = new Thread(new CalculateBotTurn(),
-                                               getName() + " Turn " + game.getTurnIndex()
-                                               + " Calc Thread"
+                            getName() + " Turn " + game.getTurnIndex() + " Calc Thread"
                     );
                     worker.start();
+                    calculatedTurnsThisPhase++;
                 }
-                
+
                 // unloading "stranded" units happens as part of a game turn change, so that's where we do it.
                 if(canUnloadStranded()) {
                     sendUnloadStranded(getStrandedEntities());
                 }
             }
 
+            @Override
+            public void gamePhaseChange(GamePhaseChangeEvent e) {
+                calculatedTurnThisPhase = false;
+                if (e.getOldPhase().isPhaseSimultaneous(game)) {
+                    int numOwnedEntities = game.getEntitiesOwnedBy(getLocalPlayer());
+                    System.out.println("BotClient calculated turns, " + getName() + " phase " + e.getOldPhase()
```

Comment:
Do we want to use that fancy pants new logging mechanism here?

Reply:
Yes, but `BotClient` doesn't use it from what I could tell.  I wanted to keep this PR tight on one specific issue.

As an aside, in looking at the logs with the new logging, it looks like it adds a lot of whitespace.   It seems like lines logged from server went from being 1 line to 3.  It looks like there's a linebreak between the "status" block of the log message, the actual message, and then a blank line after that.  I'm not a fan of this, but it seems like it's part of the logging library.  I think maybe there's a logging properties setting somewhere to change this?  I didn't have much time to look into this.

Reply:
I also think we need to adjust how we're using the logging.  I think `MMLogger` should have some static helper functions, like `logDebug` and `logWarn`, which get used with a passed-in logger.  Right now, classes that use the logging define these methods, but they're defined locally so they will end up getting repeated everywhere.

Reply:
Ok, let's leave it for now and get this merged, the rest looks fine.

Reply:
@arlith you can set the message format in the `log4j.xml` config file. There is a laundry list of options you can use, and you can pretty well format the message any way you like.